android/src/main/java/com/swmansion/rnscreens/bottomsheet/BottomSheetDialogRootView.kt

```diff
@@ -0,0 +1,102 @@
+package com.swmansion.rnscreens.bottomsheet
+
+ import android.annotation.SuppressLint
+ import android.view.MotionEvent
+ import android.view.View
+ import com.facebook.react.bridge.ReactContext
+ import com.facebook.react.config.ReactFeatureFlags
+ import com.facebook.react.uimanager.JSPointerDispatcher
+ import com.facebook.react.uimanager.JSTouchDispatcher
+ import com.facebook.react.uimanager.RootView
+ import com.facebook.react.uimanager.events.EventDispatcher
+ import com.facebook.react.views.view.ReactViewGroup
+
+ @SuppressLint("ViewConstructor")
+ class BottomSheetDialogRootView(
+    val reactContext: ReactContext?,
+    private val eventDispatcher: EventDispatcher
+ ) : ReactViewGroup(reactContext), RootView {
+    private val jsTouchDispatcher: JSTouchDispatcher = JSTouchDispatcher(this)
+    private var jsPointerDispatcher: JSPointerDispatcher? = null
+
+    init {
+        // Can we safely use ReactFeatureFlags?
```

Comment:
Let's for now remove this comment and assume we can 😄

Reply:
^Nevermind, lemme check source code of RN to find out if we can 🤔

Reply:
Unfortunately `ReactFeatureFlags` are deprecated & [the comment inside `ReactFeatureFlags.java` states as follows](https://github.com/facebook/react-native/blob/61b3c95b45b42a5b86d0a3b013b4f37d0e138613/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/config/ReactFeatureFlags.java#L14-L24): 

```
 * Hi there, traveller! This configuration class is not meant to be used by end-users of RN. It
 * contains mainly flags for features that are either under active development and not ready for
 * public consumption, or for use in experiments.
```

However new feature flag system [was added only in January](https://github.com/facebook/react-native/pull/42430) (or even later, as this commit was later reverted, so this new mechanism is not present before 0.74...

TBH I'm unsure what to do here. The situation is as follows: 

1. I need to access the feature flag here (I'm repeating after DialogRootView implementation in RN).
2. The new feature flag mechanism was introduced with 0.74
3. The old feature flag mechanism was [first deprecated with 0.74](https://github.com/facebook/react-native/commit/db6c98eac1bfc2618c0081439ab13d75f754d09f)

It would be best to have some conditional compilation here, but we're on JVM :D 

One option is I could create two separate files and include only one to source sets depending on react native version in our `build.gradle`. Basically the same as we do with `FabricEnabledViewGroup` <- on Paper we include stub implementation, on Fabric we include proper implementation.

Second option would be to stick with the deprecated mechanism for now (so leave the implementation in its current shape), cause it could take literally years before it is removed. 

Let me know what do you think.

Reply:
@kkafar Yeah, the first thing that came to my mind was how we're compiling `FabricEnabledViewGroup`, depending on the type of architecture :D
Well, since this `ReactFeatureFlags` is deprecated, I'm for the first solution - let's not use something that is meant to be removed and let's stick with adding proper source set. It would also allow us to make setting RN 0.74 as minimal supported RN version easier, when further versions of RN (we're talking about RN ~0.77 version here) will be released.